### PR TITLE
Updated Geant4 and dependencies clhep and vecgeom to use cxxstd= variant

### PR DESCRIPTION
**Pull request by gartung**
so that it is applied consistently.

### CODE REVIEW

**Comment by gartung** [comment]
@drbenmorgan can you please add a review.

File: var/spack/repos/builtin/packages/geant4/package.py [hunk 1]
```diff
@@ -19,38 +21,39 @@ class Geant4(CMakePackage):
...
+    depends_on("clhep@2.4.0.0 cxxstd=14", when="@10.04 cxxstd=14")
+    depends_on("clhep@2.3.4.6 cxxstd=14", when="@10.03.p03 cxxstd=14")
+    depends_on("vecgeom cxxstd=14", when="+vecgeom cxxstd=14")
+
+    # C++17 support
+    patch('cxx17.patch', when='@:10.03.p99 cxxstd=17')
+    patch('cxx17_geant4_10_0.patch', level=1, when='@10.04.00: cxxstd=17')
+    depends_on("clhep@2.4.0.0 cxxstd=17", when="@10.04 cxxstd=17")
+    depends_on("clhep@2.3.4.6 cxxstd=17", when="@10.03.p03 cxxstd=17")
+    depends_on("vecgeom cxxstd=17", when="+vecgeom cxxstd=17")
```
Comment (drbenmorgan) [inline var/spack/repos/builtin/packages/geant4/package.py]: With #10561 merged, and `xerces-c` always on, `depends_on('xerces-c cxxstd=NN', when='cxxstd=NN')` should be added for 11, 14 and 17.

**Comment by drbenmorgan** [comment]
@gartung only one minor suggestion in the Geant4 `package.py`. 

The data dependencies also look o.k., and definitely the way to go with this package. My only thought would be whether to split the data work into a separate PR for clarity/orthogonality, but I defer to Spack's policy here. One thing to look at would be introducing a `geant4-data` package between `geant4` and the individual data libraries. It might keep the version mapping and creation of the data "view" in one place? In `geant4` the dependency would just be expressed as

```
depends_on('geant4-data@VERSION', when='@VERSION')
```

Anyway, we can think about that later!

**Comment by gartung** [comment]
@drbenmorgan the naming of the data packages follows the FNAL ups package names. This does not always match the Geant4 name. Do you think I should rename them?

**Pull request merged**